apps/meteor/client/hooks/useRootUrlChange.tsx

```diff
@@ -0,0 +1,60 @@
+import { useEffectEvent } from '@rocket.chat/fuselage-hooks';
+import { useRole, useSetModal, useSetting, useSettingSetValue, useToastMessageDispatch } from '@rocket.chat/ui-contexts';
+import { useMutation } from '@tanstack/react-query';
+import { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import UrlChangeModal from '../components/UrlChangeModal';
+
+export const useRootUrlChange = () => {
+	const { t } = useTranslation();
+	const dispatchToastMessage = useToastMessageDispatch();
+	const isAdmin = useRole('admin');
+	const setModal = useSetModal();
+	const closeModal = useEffectEvent(() => setModal(null));
+
+	const currentUrl = location.origin + window.__meteor_runtime_config__.ROOT_URL_PATH_PREFIX;
+	const siteUrl = useSetting('Site_Url', '');
+	const documentDomain = useSetting('Document_Domain', '');
+	const setSiteUrl = useSettingSetValue('Site_Url');
+
+	const {
+		mutate: siteUrlMutation,
+		isPending,
+		isSuccess,
+	} = useMutation({
+		mutationKey: ['settings', 'Site_Url'],
+		mutationFn: async (url: string) => {
+			await setSiteUrl(url);
+			return { url };
+		},
+		onSuccess: ({ url }) => dispatchToastMessage({ type: 'success', message: `${t('Saved')}, new site url is: ${url}` }),
```

Comment:
I believe this success message needs to be 100% internationalized, not just the "Saved" 😅

Reply:
ooops, thanks